### PR TITLE
Pull optimizations to serial version

### PR DESCRIPTION
**Pull request by matt-frey**
This PR applies all code optimizations to the serial version of EPIC, these are
* option to run RK3 (see #487)
* Scherzinger eigenvalue/-vector solver (see #488 and #495)
* amax instead of vmax (see #489)
* container resizing (see #493)

### CODE REVIEW

File: src/3d/epic3d.f90 [hunk 1]
```diff
@@ -27,11 +27,12 @@ program epic3d
     use inversion_utils, only : init_inversion
     use parcel_interpl, only : grid2par_timer, par2grid_timer
     use parcel_init, only : init_timer
-    use ls_rk4, only : ls_rk4_alloc, ls_rk4_dealloc, ls_rk4_step, rk4_timer
+    use ls_rk, only : ls_rk4_alloc, ls_rk4_dealloc, ls_rk_step, rk_timer, ls_rk_setu
```
Comment (sjboeing) [inline src/3d/epic3d.f90:30]: Maybe the alloc routines should at some point ust be called ls_rk_alloc etc. But this should not matter for functionality.

**Review by sjboeing — COMMENTED** [review]
Thanks for doing this, Matthias. Do you want to set rk_order==3 for some setups or compare with/leave the code at rk_order 4?

File: src/3d/epic3d.f90 (hunk 1 above)
Comment (matt-frey) [reply src/3d/epic3d.f90]: I have just remove those routines.

**Review by sjboeing — COMMENTED** [review]
Looks good! One thing I found is that I need to remove max_vratios in order to run. Actually it would be good to make the changes that force not setting max_vratios for the parallel version as well. 

Now wondering if we may want a slightly smaller a_max to guard against changes in a within the time step to a degree. I think it is not likely to crash on this, but there may be a theoretical possibility.

**Comment by matt-frey** [comment]
Ok, thanks for letting me know. I will fix the max_vratio issue.

**Comment by matt-frey** [comment]
I removed `max_vratio.

**Review by sjboeing — APPROVED** [review]

**Pull request merged**